src/components/ThemeProvider/ThemeProviderConstants.ts

```diff
@@ -0,0 +1,51 @@
+export enum SystemTheme {
+  LIGHT = "light",
+  DARK = "dark",
+  BLACK = "black",
+  HACKER = "hacker"
+}
+
+// Colors which are eligible for theming
+export enum ThemeColor {
+  primaryColor = "primary-color",
+  primaryHoverColor = "primary-hover-color",
+  primarySelectedColor = "primary-selected-color",
+  primarySelectedHoverColor = "primary-selected-hover-color",
+  primarySelectedOnSecondaryColor = "primary-selected-on-secondary-color",
+  textColorOnPrimary = "text-color-on-primary",
+  brandColor = "brand-color",
+  brandHoverColor = "brand-hover-color",
+  brandSelectedColor = "brand-selected-color",
+  brandSelectedHoverColor = "brand-selected-hover-color",
+  textColorOnBrand = "text-color-on-brand"
+}
+
+export type Theme = {
+  name: string;
```

Comment:
I'm missing the documentation of this property and testing what happens if not passed

Reply:
Added docs and tests